drivers/espi/espi_it8xxx2.c

```diff
@@ -153,6 +155,112 @@ static const struct ec2i_t pmc1_settings[] = {
 	{HOST_INDEX_LDA, 0x01},
 };
 
+#ifdef CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD
+#define IT8XXX2_ESPI_HC_DATA_PORT_MSB \
+	((CONFIG_ESPI_PERIPHERAL_HOST_CMD_DATA_PORT_NUM >> 8) & 0xff)
+#define IT8XXX2_ESPI_HC_DATA_PORT_LSB \
+	(CONFIG_ESPI_PERIPHERAL_HOST_CMD_DATA_PORT_NUM & 0xff)
+#define IT8XXX2_ESPI_HC_CMD_PORT_MSB \
+	(((CONFIG_ESPI_PERIPHERAL_HOST_CMD_DATA_PORT_NUM + 4) >> 8) & 0xff)
+#define IT8XXX2_ESPI_HC_CMD_PORT_LSB \
+	((CONFIG_ESPI_PERIPHERAL_HOST_CMD_DATA_PORT_NUM + 4) & 0xff)
+static const struct ec2i_t pmc2_settings[] = {
+	/* Select logical device 12h(PM2 host command) */
+	{HOST_INDEX_LDN, LDN_PMC2},
+	/* I/O Port Base Address (data/command ports) */
+	{HOST_INDEX_IOBAD0_MSB, IT8XXX2_ESPI_HC_DATA_PORT_MSB},
+	{HOST_INDEX_IOBAD0_LSB, IT8XXX2_ESPI_HC_DATA_PORT_LSB},
+	{HOST_INDEX_IOBAD1_MSB, IT8XXX2_ESPI_HC_CMD_PORT_MSB},
+	{HOST_INDEX_IOBAD1_LSB, IT8XXX2_ESPI_HC_CMD_PORT_LSB},
+	/* Set IRQ=00h for logical device */
+	{HOST_INDEX_IRQNUMX, 0x00},
+	/* Enable logical device */
+	{HOST_INDEX_LDA, 0x01},
+};
+#endif
+
+#if defined(CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD) || \
+	defined(CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION)
+/*
+ * Host to RAM (H2RAM) memory mapping.
+ * This feature allows host access EC's memory directly by eSPI I/O cycles.
+ * Mapping range is 4K bytes and base address is adjustable.
+ * Eg. the I/O cycle 800h~8ffh from host can be mapped to x800h~x8ffh.
+ * Linker script of h2ram.ld will make the pool 4K aligned.
+ */
+#define IT8XXX2_ESPI_H2RAM_POOL_SIZE 0x1000
+static uint8_t h2ram_pool[IT8XXX2_ESPI_H2RAM_POOL_SIZE]
+					__attribute__((section(".h2ram_pool")));
+
+#if defined(CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION) && \
+	((CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION_PORT_NUM + \
+	CONFIG_ESPI_IT8XXX2_ACPI_SHM_H2RAM_SIZE) > IT8XXX2_ESPI_H2RAM_POOL_SIZE)
+#error "ACPI shared memory region out of h2ram"
+#endif
+
+#if defined(CONFIG_ESPI_PERIPHERAL_EC_HOST_CMD) && \
+	((CONFIG_ESPI_PERIPHERAL_HOST_CMD_PARAM_PORT_NUM + \
+	CONFIG_ESPI_IT8XXX2_HC_H2RAM_SIZE) > IT8XXX2_ESPI_H2RAM_POOL_SIZE)
+#error "EC host command parameters out of h2ram"
+#endif
+
```

Comment:
This could also use checks to verify the ACPI and HC sections don't overlap.

Reply:
Done.